### PR TITLE
Remove known issues from build cache documentation

### PR DESCRIPTION
**Pull request by wolfs**
All the issues either have an Github issue open, do not apply any more
or are documented in the build cache guide.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/buildCache.adoc [hunk 1]
```diff
@@ -191,90 +191,6 @@ We suggest adding tests to ensure that the task inputs are relocatable, i.e. tha
...
-| Actions added by a plugin (from buildSrc or externally) do not have this problem because their classloader is restricted to the classpath of the plugin.
-| Avoid adding actions to cacheable tasks in a build file.
-
-| Modifying inputs or outputs during task execution.
-| It's possible to modify a task's inputs or outputs during execution in ways that change the output of a task. This breaks incremental builds and can cause problems with the build cache.
-| Use a configure task to finalize configuration for a given task. A configure task configures another task as part of its execution.
-
-| Order of input files affects outputs.
-| Some tools are sensitive to the order of its inputs and will produce slightly different output. Gradle will usually provide the order of files from the filesystem, which will be different across operating systems.
-| Provide a stable order for tools affected by order.
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/buildCache.adoc]: I think we should mention that file order is important in some place. Not as a caveat here, but we should mention it. I can't think of a better place than `moreAboutTasks.adoc`, which in itself is a problem, but still.

Comment (lptr) [reply subprojects/docs/src/docs/userguide/buildCache.adoc]: Most of these caveats seem like interesting information. Maybe we should add them in the build cache guide?

Comment (wolfs) [reply subprojects/docs/src/docs/userguide/buildCache.adoc]: Yeah, that where I looked for the caveats. And I found them there.

E.g. the ordering thing is documented [here](https://guides.gradle.org/using-build-cache/#custom_and_third_party_tasks).

Comment (lptr) [reply subprojects/docs/src/docs/userguide/buildCache.adoc]: Ah, cool!

**Pull request merged**